### PR TITLE
Prepare release 0.62.1

### LINKED ISSUE #1176: Release the operator to use collector v0.62.1

### PR DESCRIPTION
**Pull request by frzifus**
Please have a look @Aneurysm9

Closes #1176

### CODE REVIEW

File: versions.txt
Comment (Aneurysm9) [inline versions.txt:11]:
```suggestion
targetallocator=0.62.1
```

I think this image version should be moving in parallel with the operator image.

Comment (frzifus) [reply versions.txt]: Then we might need to adapt the `0.63.1` release too!

In `0.61.0` the targetallocator was not bumped. [[diff](https://github.com/open-telemetry/opentelemetry-operator/pull/1131/files#diff-eb8281dc5868dec21668321219be72bacb9baa0acf7e43ac159d5a53e9d44abcR11)]

So i did also not touch it in `0.63.1`. Seems both passed the review. [[diff](https://github.com/open-telemetry/opentelemetry-operator/pull/1215/files#diff-eb8281dc5868dec21668321219be72bacb9baa0acf7e43ac159d5a53e9d44abcR11)]

How should we handle that? @Aneurysm9

Comment (Aneurysm9) [reply versions.txt]: v0v

I think the version of the target allocator was only recently changed to not be `0.1.0` with every release, so not sure there's a good precedent here.  I think we update it with `0.62.1` as you've done here and then going forward from `0.64.0`.

Comment (frzifus) [reply versions.txt]: But then its a upgrade in `0.62.1` and a downgrade in `0.63.1` right?
| Operator-Version  | targetallocator  |
|---|---|
| 0.60.0 | 0.60.0  |
| 0.61.0 | 0.60.0  |
| **0.62.1** | 0.62.1  |
| 0.63.1 | **0.60.0**  |
| 0.64.0 | 0.64.0  |

Comment (Aneurysm9) [reply versions.txt]: > But then its a upgrade in `0.62.1` and a downgrade in `0.63.1` right?
> ...
> 
> So would it make sense to keep `0.60.0` in the `0.62.1` release too?

No, I don't think so.  Make a `v0.36.1-oops-we-messed-up` release if you like, but I don't think it makes sense to ship old and mismatched versions of this component to maintain some notion of monotonicity that I don't think is necessary.

**Review by Aneurysm9 — APPROVED** [review]

File: versions.txt
Comment (pavolloffay) [reply versions.txt]: The TA image version is taken from https://github.com/open-telemetry/opentelemetry-operator/blob/main/versions.txt#L11 

And published only for changes is https://github.com/open-telemetry/opentelemetry-operator/blob/main/.github/workflows/publish-target-allocator.yaml#L5


but https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

> When using the push and pull_request events, you can configure a workflow to run based on what file paths are changed. Path filters are not evaluated for pushes of tags.


So it means that we can change TA version in the release PR ❤️

**Comment by pavolloffay** [comment]
I think we need to disable publishing of latest tag for the operator https://github.com/docker/metadata-action/issues/161

**Comment by pavolloffay** [comment]
After the PR is merged we should as well backport readme and changelog changes to the main branch

**Review by VineethReddy02 — APPROVED** [review]

**Comment by frzifus** [comment]
@pavolloffay could you have a look please?

**Review by pavolloffay — APPROVED** [review]
the auto-instrumentation workflows didn't have to be changed, but LGTM

**Pull request merged**
